server/handler/auth.go

```diff
@@ -19,6 +18,7 @@ func Login(oAuth *service.OAuth) http.HandlerFunc {
 
 // OAuthCallback makes exchange with oauth provider, gets&creates user and redirects to index page with JWT token
 func OAuthCallback(oAuth *service.OAuth, jwt *service.JWT, userRepo *repository.Users, uiDomain string) http.HandlerFunc {
+	logger := service.NewLogger()
```

Comment:
should it be a dependency? for example in tests you want to use another logger

Reply:
I started making the logger injectable/configurable, so you could define which one to use in different scenarios.
But I rollbacked to remove initial complexity.
Do we want that for this first iteration?

Reply:
where is a complexity to change 
```go
func OAuthCallback() {
  logger := service.NewLogger()
}
```
to
```go
func OAuthCallback(logger *logrus.FieldLogger) {
}
```
?

Reply:
It is now:
```go
func OAuthCallback(oAuth *service.OAuth, jwt *service.JWT, userRepo *repository.Users, uiDomain string) http.HandlerFunc {
```
I didn't want to add more params to some handlers, but I can do so for sure.
I saw that chi/logger uses the context to store the logger, what could be another approach.
Which one would be your option?
- a: one parammeter more,
- b: using the context as chi does

Reply:
the easiest way to just add one more param. The better way is to convert it to structure.
for logger it's okay to store it in context (but you will need a helper, which returns "default" logger if there is no logger in context), especially if you want to have request scoped logger (and I want).

But remember `Context.Value should inform, not control`. Here is a good article about it: https://medium.com/@cep21/how-to-correctly-use-context-context-in-go-1-7-8f2c0fafdf39

Reply:
so `a`, and lets proceed with `b` in the next itteration

Reply:
addressed by 28c54c4